cpp/src/arrow/compute/kernels/scalar_arithmetic_test.cc

```diff
@@ -1511,5 +1530,117 @@ TEST(TestBinaryArithmeticDecimal, Divide) {
   }
 }
 
+TYPED_TEST(TestUnaryArithmeticFloating, TrigSin) {
+  this->SetNansEqual(true);
+  for (auto check_overflow : {false, true}) {
+    this->SetOverflowCheck(check_overflow);
+    this->AssertUnaryOp(Sin, "[]", "[]");
+    this->AssertUnaryOp(Sin, "[null, NaN]", "[null, NaN]");
+    this->AssertUnaryOp(Sin, MakeArray(0, M_PI_2, M_PI), "[0, 1, 0]");
+  }
+  this->AssertUnaryOpRaises(Sin, "[Inf, -Inf]", "domain error");
```

Comment:
This is for the checked variant, but what happens for the non-checked variant? Return `NaN`?

Reply:
Ping on this comment.

Reply:
Sorry - yes, it returns NaN. I added an assertion above but forgot to comment here.